### PR TITLE
hal: fix http request encode error

### PR DESCRIPTION
**Pull request by oguzdemirbasci**

### CODE REVIEW

File: backend/inspirehep/hal/core/sword.py [hunk 1]
```diff
@@ -92,4 +92,4 @@ def _create_payload(tei, doc_file):
 
         return temp_file, "application/zip", "meta.xml"
 
-    return tei, "text/xml", "meta.xml"
+    return tei.encode("utf-8"), "text/xml", "meta.xml"
```
Comment (drjova) [inline backend/inspirehep/hal/core/sword.py:95]: could we add a test for that?

Comment (drjova) [reply backend/inspirehep/hal/core/sword.py:95]: @oguzdemirbasci  :)

Comment (oguzdemirbasci) [reply backend/inspirehep/hal/core/sword.py:95]: Tested with a record that had given the encoding error on QA before. VCR casette generated.

File: backend/tests/integration/hal/test_core_sword.py [hunk 2]
```diff
@@ -100,3 +100,23 @@ def test_push(app, get_fixture):
...
+def test_unicode_data(app, get_fixture):
+    record_json = orjson.loads(get_fixture("hal_preprod_unicode_record.json"))
+    record_data = faker.record("lit", data=record_json)
+    record = InspireRecord.create(record_data)
+
+    institute_json = orjson.loads(get_fixture("hal_preprod_unicode_institute.json"))
+    institute_data = faker.record("ins", data=institute_json)
+    institute_record = InspireRecord.create(institute_data)
+
+    # HAL Create
```
Comment (drjova) [inline backend/tests/integration/hal/test_core_sword.py:115]: unnecessary comment :)

**Review by drjova — APPROVED** [review]
Just the comment 👍

**Pull request merged**